AutofacContrib.NSubstitute/AutoSubstituteBuilder.cs

```diff
@@ -48,6 +55,18 @@ public AutoSubstituteBuilder ConfigureBuilder(Action<ContainerBuilder> action)
             return this;
         }
 
+        /// <summary>
+        /// Allows for decorating of the container that results from the call to <see cref="Build"/>
+        /// </summary>
+        /// <param name="action">A delegate to run to configure the container.</param>
+        /// <returns>The current <see cref="AutoSubstituteBuilder"/>.</returns>
+        public AutoSubstituteBuilder ConfigureContainer(Func<IContainer, IContainer> action)
```

Comment:
One last thing on this one - do you think it makes sense to have a `ConfigureContainer`, there isn't a whole lot that can be done on the `IContainer` interface. Is there anything apart from providing a container you'd want to use?

Perhaps we should just do a `WithContainer` method?

Also, the usage style here where the container gets replaced means that the `.RegisterSource` calls and the `provided.SetContainer` calls won't affect that container, which might be surprising behaviour for the consumer. Thoughts?

Reply:
Maybe a better approach is to allow a `ConfigureContainerBuilder` method so you can set up the container builder how you want to get the desired container?

Reply:
Good points! I'll look into that

Reply:
I've updated the provided values are set via a list of after-build actions, and the builder is not created with a `ConfigureContainerBuilder`.